src/app/api/common/delegates/getDelegates.ts

```diff
@@ -61,7 +62,7 @@ export async function getDelegates({
             take
           );
         default:
-          return prisma[`${namespace}Delegates`].findMany({
+          return (prisma as any)[`${namespace}Delegates`].findMany({
```

Comment:
don't love this but don't have a better suggesting atm

I think the typechecks are generally helpful. 

One option could be creating ghost views in Prostgres / prisma

Reply:
@stepandel same. I briefly looked into ways of abstracting this further but there wasn't an easy solution. I like the idea of a generic ghost view though.